modules/imgproc/src/lsd.cpp

```diff
@@ -430,6 +430,9 @@ void LineSegmentDetectorImpl::detect(InputArray _image, OutputArray _lines,
     if(w_needed) Mat(w).copyTo(_width);
     if(p_needed) Mat(p).copyTo(_prec);
     if(n_needed) Mat(n).copyTo(_nfa);
+
+    // Clear used structures
+    list.clear();
```

Comment:
It looks like `list` vector isn't used in any meaningful way. IMHO it could be completely removed.

Reply:
What do you mean?
It is used in other methods:
- ll_angle()
- flsd()

Reply:
`ll_angle` fills it as a row/column ordered list of all image points(actually almost all) and link it in special order, however the list head and tail pointers are local variables and are discarded as soon as list is constructed and the link order isn't used anywhere also.
`flsd` function use the list as an array to iterate over all stored points in a straight way and doesn't consider list ordering. So actually it just iterates over all image points.